v2/api/containerservice/customizations/managed_cluster_extensions.go

```diff
@@ -120,3 +124,59 @@ func secretsToWrite(obj *containerservice.ManagedCluster, adminCreds string, use
 
 	return collector.Values()
 }
+
+var _ extensions.PreReconciliationChecker = &ManagedClusterExtension{}
+
+// If a managed cluster has a provisioningState not in this set, it will reject any attempt to PUT a new state out of
+// hand; so there's no point in even trying. This is true even if the PUT we're doing will have no effect on the state
+// of the cluster.
+// These are all listed lowercase, so we can do a case-insensitive match.
+var nonBlockingManagedClusterProvisioningStates = set.Make(
+	"succeeded",
+	"failed",
+	"canceled",
+	"ready",
```

Comment:
FWIW I think it'd probably be safe to not have this here as AKS doesn't have the same ARM contract violation as SQL.